server/routes/login.js

```diff
@@ -1,207 +1,33 @@
-        `${userAuthentication.ApiUrl}health`,
-        `${userAuthentication.ApiUrl}health/invision`,
-      ])
-      isApiUp = healthRes.status === 200
-      log.info(`loginIndex - health check called and the isAppUp = ${isApiUp} with status ${healthRes.status}`)
-      log.info(`${userAuthentication.ApiUrl}health - health check with status ${healthRes.status}`)
-
-      if (isApiUp === false) {
-        log.error(healthRes.appInfo)
-        res.render('pages/index', {
-          showUserNotSignedUpMessage: false,
-          authError: false,
-          apiUp: isApiUp,
-          csrfToken: res.locals.csrfToken,
-        })
-        return
-      }
-
-      const quantumAddresses = config.quantumAddresses.split(',')
-
-      if (config.twoFactorAuthOn === 'true' && ipRangeCheck(ipAddress, quantumAddresses) === false) {
-        if (userAuthenticationDetails === null || userAuthenticationDetails.length === 0) {
-          throw new Error(`Error : No Sms or Email address returned for QuantumId : ${req.user.username}`)
-        }
-
-        // eslint-disable-next-line no-shadow
-        const userAuthentication = userAuthenticationDetails[0]
-
-        if (
-          (userAuthentication.EmailAddress === null || userAuthentication.EmailAddress === '') &&
-          (userAuthentication.Sms === null || userAuthentication.Sms === '')
-        ) {
-          throw new Error(`Error : Sms or Email address null or empty for QuantumId : ${req.user.username}`)
-        }
-
-        const emailEnabled = userAuthentication.UseEmailAddress
-        const smsEnabled = userAuthentication.UseSms
-
-        if (!emailEnabled && !smsEnabled) {
-          throw new Error(`Error : Sms or Email address both set to false for QuantumId : ${req.user.username}`)
-        }
-
-        const twofactorCode = utilities.get2faCode()
-
-        await userAuthenticationService.updateTwoFactorAuthenticationHash(
-          req.user.username,
-          utilities.createTwoFactorAuthenticationHash(twofactorCode.toString()),
-        )
-
-        if (smsEnabled) {
-          // For SMS
-          await notify
-            .sendSms(notifySmsTemplate, userAuthentication.Sms || '', {
-              personalisation: { '2fa_code': twofactorCode },
-            })
-            .catch((err) => {
-              throw new Error(err)
-            })
-        }
-
-        if (emailEnabled) {
-          // For email
-          await notify
-            .sendEmail(notifyEmailTemplate, userAuthentication.EmailAddress || '', {
-              personalisation: { '2fa_code': twofactorCode },
-            })
-            .catch((err) => {
-              throw new Error(err)
-            })
-        }
-
+      const userAuthenticationDetails = userAuthenticationService.getUserAuthenticationDetails(req.user.username)
+
+      if (
+        userHasRole(req.user.token, 'ROLE_MFA') === false &&
+        isQuantumAddress(req) === false &&
+        config.twoFactorAuthOn === 'true'
```

Comment:
If we're just checking for falsey results we don't need the _=== false_ 

```
if( !userHasRole(req.user.token, 'ROLE_MFA') &&
!isQuantumAddress(req) &&
config.twoFactorAuthOn === 'true')
```

Reply:
`!` also returns true for all falsy values so we would get a truthy return if returned `null`. Not looked at the function so not sure if this is possible

Reply:
I'd probably also be tempted to convert the _config. twoFactorAuthOn_ to a Boolean in config too...

Reply:
And provided we don't lose context, we could just extract all the config params on require. Certainly this one stands alone. I think the context may be useful elseware though.